server/lib/src/idm/event.rs

```diff
@@ -276,6 +276,26 @@ impl LdapTokenAuthEvent {
     }
 }
 
+pub struct LdapApplicationAuthEvent {
+    pub application: String,
+    pub target: Uuid,
+    pub cleartext: String,
+}
+
+impl LdapApplicationAuthEvent {
+    pub fn from_parts(
+        app_name: &str,
+        usr_uuid: Uuid,
+        cleartext: String,
+    ) -> Result<Self, OperationError> {
+        Ok(LdapApplicationAuthEvent {
+            application: app_name.to_string(),
+            target: usr_uuid,
+            cleartext,
+        })
+    }
+}
```

Comment:
Shouldn't this just be a `new` ?

```suggestion
impl LdapApplicationAuthEvent {
    pub fn new(
        app_name: impl ToString,
        usr_uuid: Uuid,
        cleartext: String,
    ) -> Result<Self, OperationError> {
        Ok(LdapApplicationAuthEvent {
            application: app_name.to_string(),
            target: usr_uuid,
            cleartext,
        })
    }
}
```